src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/XGBoostEvidenceFilter.java

```diff
@@ -0,0 +1,609 @@
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.TextCigarCodec;
+import htsjdk.tribble.AbstractFeatureReader;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.*;
+import org.broadinstitute.hellbender.tools.spark.utils.IntHistogram;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import scala.Tuple2;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * A class that acts as a filter for BreakpointEvidence.
+ * Features are calculated according to evidence type, overlap information, mapping quality, etc.
+ * A trained classifier scores the probability the evidence overlaps a breakpoint interval, and passes evidence above
+ * the specified threshold.
+ */
+public final class XGBoostEvidenceFilter implements Iterator<BreakpointEvidence> {
+    static private final boolean useFastMathExp = true;
+    private static final List<String> defaultEvidenceTypeOrder = Arrays.asList(
+            "TemplateSizeAnomaly",
+            "MateUnmapped", "InterContigPair",
+            "SplitRead", "LargeIndel", "WeirdTemplateSize", "SameStrandPair", "OutiesPair"
+    );
+
+    static private final int NUM_OVERLAP = 0;
+    static private final int NUM_COHERENT = 1;
+    static private final int OVERLAP_MAPPING_QUALITY = 2;
+    static private final int COHERENT_MAPPING_QUALITY = 3;
+
+    private final PartitionCrossingChecker partitionCrossingChecker;
+
+    private final Predictor predictor;
+    private final Map<String, Integer> evidenceTypeMap;
+    private final double coverage;
+    private final double thresholdProbability;
+    private final Map<String, String> readGroupToLibraryMap;
+    private final Map<String, LibraryStatistics> libraryStatisticsMap;
+
+    private final EvidenceOverlapChecker evidenceOverlapChecker;
+    private final Map<BreakpointEvidence, int[]> rawFeatureCache;
+
+    private Iterator<SVIntervalTree.Entry<List<BreakpointEvidence>>> treeItr;
+    private Iterator<BreakpointEvidence> listItr;
+    private final SVIntervalTree<Double> genomeGaps;
+    private final SVIntervalTree<Double> umapS100Mappability;
+
+    public XGBoostEvidenceFilter(
+            final Iterator<BreakpointEvidence> evidenceItr,
+            final ReadMetadata readMetadata,
+            final StructuralVariationDiscoveryArgumentCollection.FindBreakpointEvidenceSparkArgumentCollection params,
+            final PartitionCrossingChecker partitionCrossingChecker
+    ) {
+        this.predictor = loadPredictor(params.svEvidenceFilterModelFile);
+        this.evidenceTypeMap = loadEvidenceTypeMap(params.svCategoricalVariablesFile);
+        this.genomeGaps = new SVIntervalTree<>();
+        loadGenomeIntervals(this.genomeGaps, params.svGenomeGapsFile, readMetadata);
```

Comment:
I changed this to use FeatureDataSource as suggested by you in a later comment, removing most of this code. Many subsequent criticisms of this design are noted for future reference but not responded to because they're now moot.